automerge-wasm/README.md

```diff
@@ -1,696 +1,478 @@
+
+A very simple sync implementation might look like this.
+
+```javascript
+  import { encodeSyncState, decodeSyncState, initSyncState } from "automerge-wasm"
+
+  let states = {}
+
+  function receiveMessageFromPeer(doc, peer_id, message) {
+      let syncState = states[peer_id]
+      doc.receiveMessage(syncState, message)
+      let reply = doc.generateSyncMessage(syncState)
+      if (reply) {
+          sendMessage(peer_id, reply)
+      }
+  }
+
+  function notifyPeerAboutUpdates(doc, peer_id) {
+      let syncState = states[peer_id]
+      let message = doc.generateSyncMessage(syncState)
+      if (message) {
+          sendMessage(peer_id, message)
+      }
+  }
+
+  function onDisconnect(peer_id) {
+      let state = states[peer_id]
+      if (state) {
+        saveSyncToStorage(peer_id, encodeSyncState(state))
+      }
+      delete states[peer_id]
+  }
+
+  function onConnect(peer_id) {
+      let state = loadSyncFromStorage(peer_id)
+      if (state) {
+        states[peer_id] = decodeSyncState(state)
+      } else {
+        states[peer_id] = initSyncState()
+      }
+  }
+```
+
+### Glossary: Actors
+
+Some basic concepts you will need to know to better understand the api are Actors and Object Ids.
+
+Actors are ids that need to be unique to each process writing to a document.  This is normally one actor per device.  Or for a web app one actor per tab per browser would be needed.  It can be a uuid, or a public key, or a certificate, as your application demands.  All that matters is that its bytes are unique.  Actors are always expressed in this api as a hex string.
+
+Methods that create new documents will generate random actors automatically - if you wish to supply your own it is always taken as an optional argument.  This is true for the following functions.
+
+```javascript
+  import { create, load } from "automerge-wasm"
+
+  let doc1 = create()  // random actorid
+  let doc2 = create("aabbccdd")
+  let doc3 = doc1.fork()  // random actorid
+  let doc4 = doc2.fork("ccdd0011")
+  let doc5 = load(doc3.save()) // random actorid
+  let doc6 = load(doc4.save(), "00aabb11")
+
+  let actor = doc1.getActor()
+
+  doc1.free(); doc2.free(); doc3.free(); doc4.free(); doc5.free(); doc6.free()
+```
+
+### Glossary: Object Id's
+
+Object Ids uniquely identify an object within a document.  They are represented as strings in the format of `{counter}@{actor}`.  The root object is a special case and can be referred to as `_root`.  The counter is an ever increasing integer, starting at 1, that is always one higher than the highest counter seen in the document thus far.  Object Id's do not change when the object is modified but they do if it is overwritten with a new object.
```

Comment:
Perhaps worth noting that using the old object id still accesses the old object though it is no longer visible in the document?